### PR TITLE
fix: Ignore HTTP store client errors trying to get content length

### PR DESCRIPTION
**Pull request by vrongmeal**
The `HEAD` method might not be supported. Ignore client side errors and assume we have `0` content length.

Fixes: #2828

### CODE REVIEW

File: crates/datasources/src/object_store/http.rs [hunk 1]
```diff
@@ -326,10 +334,19 @@ impl ObjectStore for SimpleHttpStore {
             if content_length != 0 {
                 self.obj_store.get_opts(location, options).await
             } else {
-                self.simple_get_req(location.clone()).await.map_err(|e| {
+                self.simple_get_req(location.clone()).await.map_err(|err| {
+                    let err = if self.url.as_str().contains('*') {
+                        ObjectStoreSourceError::String(format!(
+                            "{err}\n\
+                            Note that globbing is not supported for HTTP.",
```
Comment (tychoish) [inline crates/datasources/src/object_store/http.rs:341]: I would avoid newlines in errors.

File: testdata/sqllogictests/functions/json_scan.slt [hunk 2]
```diff
@@ -40,11 +40,6 @@ select * from ndjson_scan([]);
 
 # Glob patterns not supported on HTTP
 
-statement error Unexpected status code '404 Not Found'
-select * from ndjson_scan(
-  'https://raw.githubusercontent.com/GlareDB/glaredb/main/testdata/sqllogictests_datasources_common/data/*.ndjson'
```
Comment (tychoish) [inline testdata/sqllogictests/functions/json_scan.slt:45]: why did this error before and doesn't any more?

**Review by tychoish — COMMENTED** [review]
nits

File: crates/datasources/src/object_store/http.rs (hunk 1 above)
Comment (vrongmeal) [reply crates/datasources/src/object_store/http.rs:341]: The error is usually very long. Moreover, we just want to append a "note" here for the user (easier to read).

File: testdata/sqllogictests/functions/json_scan.slt (hunk 2 above)
Comment (vrongmeal) [reply testdata/sqllogictests/functions/json_scan.slt:45]: So the previous error had both: status code and glob. Now the error message has changed...

File: crates/datasources/src/object_store/http.rs (hunk 1 above)
Comment (tychoish) [reply crates/datasources/src/object_store/http.rs:341]: I mean, I think it's fine.

If it's very long it's going to wrap anyway and adding our own new line character isn't going to actually be a great place in all cases.

My concern is that it's going to end up in log messages with a new line that's going to be hard to read and maybe hard to parse. 

I'd also change it to put the summary at the beginning: `HTTP sources do not support globbing: {err}`

**Comment by vrongmeal** [comment]
A test fails with a less precise error, should we just let it be for now. There isn't a better work around to improve the error message...

**Comment by talagluck** [comment]
It looks like we're removing tests here - are there any tests we could add?

**Comment by vrongmeal** [comment]
> It looks like we're removing tests here - are there any tests we could add?

So, to make most of the things work, we had to do a bit of hacky stuff which messes up the error messages. So the tests removed are simply not checking some particular error messages (related to HTTP). We still have the same tests below checking another error message so our coverage is still the same.

**Review by tychoish — APPROVED** [review]
